libcrux-ml-dsa/src/simd/portable/arithmetic.rs

```diff
@@ -6,60 +6,53 @@ use crate::{
     },
 };
 
+#[cfg(hax)]
+use crate::simd::traits::{add_post, add_pre, sub_post, sub_pre, COEFFICIENTS_IN_SIMD_UNIT};
+
 pub(crate) const MONTGOMERY_SHIFT: u8 = 32;
 
 #[inline(always)]
 #[hax_lib::fstar::options("--z3rlimit 150")]
-#[hax_lib::requires(fstar!(r#"forall i. i < 8 ==> 
-    Spec.Utils.is_intb (pow2 31 - 1) (v (Seq.index ${lhs}.f_values i) + v (Seq.index ${rhs}.f_values i))"#))]
-#[hax_lib::ensures(|result| fstar!(r#"forall i. i < 8 ==>
-    (v (Seq.index ${lhs}_future.f_values i) == 
-     v (Seq.index ${lhs}.f_values i) + v (Seq.index ${rhs}.f_values i))"#))]
+#[hax_lib::requires(add_pre(&lhs.values, &rhs.values))]
+#[hax_lib::ensures(|result| add_post(&lhs.values, &rhs.values, &(future(lhs).values)))]
```

Comment:
Sounds a good idea. I will try to apply that where I find a condition match with functions of `Operations` trait.

Reply:
Yes, more generally, whenever we see a formula that is being repeated in multiple places, we can now define it once and use it multiple times, using the new Prop features provided by https://github.com/cryspen/hax/pull/1301